tool/src/main/java/org/datacommons/tool/Processor.java

```diff
@@ -4,23 +4,40 @@
 import java.io.File;
 import java.io.IOException;
 import java.util.List;
+import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.datacommons.proto.Debug;
 import org.datacommons.proto.Mcf;
+import org.datacommons.util.LogWrapper;
 import org.datacommons.util.McfParser;
 import org.datacommons.util.McfUtil;
 import org.datacommons.util.TmcfCsvParser;
 
+class DCTooManyFailuresException extends Exception {
+  public DCTooManyFailuresException() {}
+
+  public DCTooManyFailuresException(String message) {
+    super(message);
+  }
+}
+
 public class Processor {
-  public static void processNodes(
-      Mcf.McfType type, File file, Debug.Log.Builder logCtx, Logger logger) throws IOException {
-    int numNodesProcessed = 0;
+  private static final Logger logger = LogManager.getLogger(Processor.class);
+
+  public static void processNodes(Mcf.McfType type, File file, Debug.Log.Builder logCtx)
+      throws IOException, DCTooManyFailuresException {
+    long numNodesProcessed = 0;
     logger.debug("Checking {}", file.getName());
     // TODO: isResolved is more allowing, be stricter.
-    McfParser parser = McfParser.init(type, file.getPath(), false);
+    McfParser parser = McfParser.init(type, file.getPath(), false, logCtx);
```

Comment:
I wonder should always put ctx as the first argument (Golang does this..)

Reply:
these could also be non-static classes, and carry the log context as class state.

Reply:
yes, good point, I was making the change in follow-on commits... folded it into here.